test/test_connection.py

```diff
@@ -58,3 +58,21 @@ def test_mock(self):
             pass
 
         self.assertEqual(original, self.conn.backend)
+
+    def test_list(self):
```

Comment:
Here I think it's probably sufficient to ensure that it calls the appropriate backend method correctly. But I don't object to being thorough like this :-)